tests/unittests/sources/test_nocloud.py

```diff
@@ -3,36 +3,31 @@
 import os
 import textwrap
 
+import pytest
 import yaml
 
-from cloudinit import dmi, helpers, util
+from cloudinit import helpers
 from cloudinit.sources.DataSourceNoCloud import DataSourceNoCloud as dsNoCloud
 from cloudinit.sources.DataSourceNoCloud import parse_cmdline_data
-from tests.unittests.helpers import CiTestCase, ExitStack, mock, populate_dir
+from tests.unittests.helpers import mock, populate_dir
 
 
-@mock.patch("cloudinit.sources.DataSourceNoCloud.util.is_lxd")
-class TestNoCloudDataSource(CiTestCase):
-    def setUp(self):
-        super(TestNoCloudDataSource, self).setUp()
-        self.tmp = self.tmp_dir()
-        self.paths = helpers.Paths(
-            {"cloud_dir": self.tmp, "run_dir": self.tmp}
-        )
+@pytest.fixture
+def paths(tmpdir):
+    yield helpers.Paths({"cloud_dir": tmpdir, "run_dir": tmpdir})
```

Comment:
We already have a paths fixture in the [top-level conftest.py](https://github.com/canonical/cloud-init/blob/main/conftest.py#L201) so no need to reinvent here.